backend/services/mgmt/v1alpha1/connection-service/connection.go

```diff
@@ -163,6 +163,51 @@ func (s *Service) CheckConnectionConfig(
 	}
 }
 
+func (s *Service) CheckConnectionConfigById(
+	ctx context.Context,
+	req *connect.Request[mgmtv1alpha1.CheckConnectionConfigByIdRequest],
+) (*connect.Response[mgmtv1alpha1.CheckConnectionConfigByIdResponse], error) {
+	connectionUuid, err := nucleusdb.ToUuid(req.Msg.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := s.db.Q.GetConnectionById(ctx, s.db.Db, connectionUuid)
+	if err != nil && !nucleusdb.IsNoRows(err) {
+		return nil, err
+	} else if err != nil && nucleusdb.IsNoRows(err) {
+		return nil, nucleuserrors.NewNotFound("unable to find connection by id")
+	}
+
+	_, err = s.verifyUserInAccount(ctx, nucleusdb.UUIDString(conn.AccountID))
+	if err != nil {
+		return nil, err
+	}
+
+	dtoVal, err := dtomaps.ToConnectionDto(&conn)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
All of this logic can be removed and replaced with an internal call to `GetConnection`. 

Then this function just becomes:

```
connResp, err := s.GetConnection(ctx, ...)
if err != nil {
  return nil, err
}

... your other check code below ...
```